internal/keepers/service.go

```diff
@@ -178,37 +181,31 @@ func (s *onDemandUpkeepService) processLatestHead(ctx context.Context) {
 
 	// Get only the active upkeeps from the contract. This should not include
 	// any cancelled upkeeps.
-	keys, err := s.registry.GetActiveUpkeepKeys(ctx, "0")
+	blockKey, keys, err := s.registry.GetActiveUpkeepKeys(ctx, "0")
 	if err != nil {
-		s.samplingResults.purge()
 		s.logger.Printf("%s: failed to get upkeeps from registry for sampling", err)
 		return
 	}
 
 	s.logger.Printf("%d active upkeep keys found in registry", len(keys))
-	if len(keys) == 0 {
-		s.samplingResults.purge()
-		return
-	}
 
 	// select x upkeeps at random from set
 	keys = s.shuffler.Shuffle(keys)
-	size := s.ratio.OfInt(len(keys))
+	sampleSize := s.ratio.OfInt(len(keys))
 
-	s.logger.Printf("%d results selected by provided ratio %s", size, s.ratio)
-	if size <= 0 {
-		s.samplingResults.purge()
```

Comment:
As part of the offline discussion with @infiloop2, we decided to remove the purging of the sampling results. Just making a comment here as a reminder to discuss if we need to keep the purging as-is.